vm/systemSmartContracts/validator.go

```diff
@@ -2242,9 +2242,9 @@ func (v *validatorSC) moveStakeFromValidatorToValidator(args *vmcommon.ContractC
 		return returnCode
 	}
 
-	validatorA := args.Arguments[0]
-	validatorB := args.Arguments[1]
-	stakeToMove := big.NewInt(0).SetBytes(args.Arguments[2])
+	validatorA := args.Arguments[1]
+	validatorB := args.Arguments[2]
+	stakeToMove := big.NewInt(0).SetBytes(args.Arguments[0])
```

Comment:
nice catch! 

TestDelegationManagerSystemSC_ChangeStakingProvider seems not to test the happy path correctly if it worked with the arguments misplaced.

Reply:
that is strange - as the happy path seems to do what the new code wants ==))

Reply:
ah. not strange - as the tests were fixed as well.